File: be-hospital/src/main/java/com/example/be_hospital/service/MedicineService.java
package com.example.be_hospital.service;

import com.example.be_hospital.dto.medicine.MedicineDTO;
import com.example.be_hospital.dto.ResponseObject;

import java.util.List;

public interface MedicineService {
    List<MedicineDTO> getAllMedicines();
    MedicineDTO getMedicineById(int id);
    ResponseObject createMedicine(MedicineDTO dto);
    ResponseObject updateMedicine(int id, MedicineDTO dto);
    ResponseObject deleteMedicine(int id);
}

File: fe-hospital/src/services/medicine.service.ts
import api from './api';

export interface MedicineDTO {
  id?: number;
  name: string;
  unit: string;
  active: boolean;
}

export interface ResponseObject {
  status: boolean;
  message: string;
}

export const medicineService = {
  getAllMedicines: async (): Promise<MedicineDTO[]> => {
    const response = await api.get<MedicineDTO[]>('/medicines');
    return response.data;
  },

  getMedicineById: async (id: number): Promise<MedicineDTO> => {
    const response = await api.get<MedicineDTO>(`/medicines/${id}`);
    return response.data;
  },

  createMedicine: async (dto: MedicineDTO): Promise<ResponseObject> => {
    const response = await api.post<ResponseObject>('/medicines', dto);
    return response.data;
  },

  updateMedicine: async (id: number, dto: MedicineDTO): Promise<ResponseObject> => {
    const response = await api.put<ResponseObject>(`/medicines/${id}`, dto);
    return response.data;
  },

  deleteMedicine: async (id: number): Promise<ResponseObject> => {
    const response = await api.delete<ResponseObject>(`/medicines/${id}`);
    return response.data;
  },
};
